src/main/java/org/commcare/formplayer/objects/QueryData.java

```diff
@@ -40,6 +41,17 @@ public void setForceManualSearch(String key, Boolean value) {
         setProperty(key, value, KEY_FORCE_MANUAL_SEARCH);
     }
 
+    public String getInitiatedBy(String key) {
+        Map<String, Object> value = (Map<String, Object>) this.get(key);
+        if (value != null) {
+            String initiatedBy = (String) value.get(INITIATED_BY);
+            if (initiatedBy != null) {
```

Comment:
nit: This null check seems unnecessary to me since you will return `null` anyway if `initiatedBy` is `null`.  maybe it's more readable with the check?

Reply:
yes you're right [e943ab5](https://github.com/dimagi/formplayer/pull/1533/commits/e943ab577db1c703cfdd8471442e9fb98937e7cc)